Sources/Swift/Integrations/SessionReplay/SentrySessionReplay.swift

```diff
@@ -268,14 +273,17 @@ class SentrySessionReplay: NSObject {
         }
         .compactMap(breadcrumbConverter.convert(from:))
     }
-
+    
     private func takeScreenshot() {
         guard let rootView = rootView, !processingScreenshot else { return }
-
-        lock.synchronized {
-            guard !processingScreenshot else { return }
-            processingScreenshot = true
+ 
+        lock.lock()
+        guard !processingScreenshot else {
+            lock.unlock()
+            return
         }
+        processingScreenshot = true
```

Comment:
Shouldn't we also be synchronizing access to where we set `processingScreenshot = false` on line 296 below, since `newImage(...)` is called from this function?